x-pack/test/functional/apps/canvas/custom_elements.ts

```diff
@@ -0,0 +1,164 @@
+      });
+    });
+
+    it('saves custom element modifications', async () => {
+      await PageObjects.canvas.openAddElementModal();
+
+      // open the custom elements tab
+      const elem = await find.byCssSelector('#customElements', 20000);
+      await elem.click();
+
+      // ensure the correct amount of custom elements exist
+      const container = await find.byCssSelector('[aria-labelledby="customElements"]');
+      const customElements = await container.findAllByCssSelector('.canvasElementCard__wrapper');
+      expect(customElements).to.have.length(1);
+
+      // hover over the custom element to bring up the edit and delete icons
+      const customElement = customElements[0];
+      await customElement.moveMouseTo();
+
+      // click the edit element button
+      const editBtn = await customElement.findByCssSelector('[aria-label="Edit element"]');
+      await editBtn.click();
+
+      // fill out the custom element form and submit it
+      await PageObjects.canvas.fillOutCustomElementForm(
+        'My Edited New Element',
+        'An excellent edited element'
+      );
+
+      // ensure the custom element in the modal shows the updated text
+      await retry.try(async () => {
+        const customElementAgain = await find.byCssSelector(
+          '[aria-labelledby="customElements"] .canvasElementCard__wrapper'
+        );
+
+        const elementName = await customElementAgain.findByCssSelector('.euiCard__title');
+        expect(await elementName.getVisibleText()).to.contain('My Edited New Element');
+      });
+    });
+
+    it('deletes custom element when prompted', async () => {
+      // open the custom elements tab
+      const elem = await find.byCssSelector('#customElements', 20000);
+      await elem.click();
+
+      // ensure the correct amount of custom elements exist
+      const container = await find.byCssSelector('[aria-labelledby="customElements"]');
+      const customElements = await container.findAllByCssSelector('.canvasElementCard__wrapper');
+      expect(customElements).to.have.length(1);
+
+      // hover over the custom element to bring up the edit and delete icons
+      const customElement = customElements[0];
+      await customElement.moveMouseTo();
+
+      // click the delete element button
+      const editBtn = await customElement.findByCssSelector('[aria-label="Delete element"]');
+      await editBtn.click();
+
+      await testSubjects.click('confirmModalConfirmButton');
+
+      // ensure the custom element was deleted
+      await retry.try(async () => {
+        const containerAgain = await find.byCssSelector('[aria-labelledby="customElements"]');
+        const customElementsAgain = await containerAgain.findAllByCssSelector(
+          '.canvasElementCard__wrapper'
+        );
+        expect(customElementsAgain).to.have.length(0);
+      });
+
+      // Close the modal
+      await browser.pressKeys(browser.keys.ESCAPE);
+    });
```

Comment:
It is not critical, but nice to wrap element search and interactions into "user-readable" functions. In this case, test scenario is easy to understand and support in the future.

```
it('deletes custom element when prompted', async () => {
  await PageObjects.canvas.openCustomElementsTab();
  await PageObjects.canvas.checkCustomElementsCount(1);
  await PageObjects.canvas.deleteElementByIndex(0);
  await PageObjects.canvas.checkCustomElementsCount(0);
}
```